src/realm/array_decimal128.hpp

```diff
@@ -40,7 +40,7 @@ class ArrayDecimal128 : public ArrayPayload, private Array {
 
     static Decimal128 default_value(bool nullable)
     {
-        return nullable ? Decimal128() : Decimal128(0);
+        return nullable ? Decimal128(realm::null()) : Decimal128(0);
```

Comment:
Since the conversion is implicit, you could just do `nullable ? null() : Decimal128(0)`. Whichever you find clearer.

Reply:
I'll prefer to be explicit and use the constructor here.